tablemusthave/musthave.py

```diff
@@ -30,10 +58,14 @@ def description(self):
     def check(self, t):
         missing = [c for c in self.colnames if c not in t]
         return must_have_result(missing=missing)
+    
+    def fix(self, t):
+        self.fix_fn(t, self.colnames)
 
 class columns_matching:
-    def __init__(self, pattern):
+    def __init__(self, pattern, fix_fn = lambda t: t):
```

Comment:
This is clever, to inject the fix method in the constructor. I wonder if you really need this level of flexibility, though. How often are you passing custom `fix_fn`'s to `columns_matching` and other constructors? Is there no way to write a normal method for each class that's flexible enough to handle the fixes for that class? If you need to override in rare cases, why not go full-on monkeypatching and replace the fix() method of the instance? Even here, I could see where the approach might not be flexible *enough*--a custom fix method for `columns_matching` might want to have access to the `pattern` attribute, for instance.

You don't have to answer all these questions, just think about it. If you ultimately decide this is the right way to go, let's do it.